pkg/artifactory/resource_artifactory_ldap_setting.go

```diff
@@ -0,0 +1,286 @@
+package artifactory
+
+import (
+	"context"
+	"encoding/xml"
+	"github.com/go-resty/resty/v2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"gopkg.in/yaml.v2"
+)
+
+type LdapSetting struct {
+	Key                      string         `xml:"key" yaml:"key"`
+	Enabled                  bool           `xml:"enabled" yaml:"enabled"`
+	LdapUrl                  string         `xml:"ldapUrl" yaml:"ldapUrl"`
+	UserDnPattern            string         `xml:"userDnPattern" yaml:"userDnPattern"`
+	EmailAttribute           string         `xml:"emailAttribute" yaml:"emailAttribute"`
+	AutoCreateUser           bool           `xml:"autoCreateUser" yaml:"autoCreateUser"`
+	LdapPoisoningProtection  bool           `xml:"ldapPoisoningProtection" yaml:"ldapPoisoningProtection"`
+	AllowUserToAccessProfile bool           `xml:"allowUserToAccessProfile" yaml:"allowUserToAccessProfile"`
+	PagingSupportEnabled     bool           `xml:"pagingSupportEnabled" yaml:"pagingSupportEnabled"`
+	Search                   LdapSearchType `xml:"search" yaml:"search"`
+}
+
+type LdapSearchType struct {
+	SearchSubTree   bool   `yaml:"searchSubTree" xml:"searchSubTree"`
+	SearchFilter    string `yaml:"searchFilter" xml:"searchFilter"`
+	SearchBase      string `yaml:"searchBase" xml:"searchBase"`
+	ManagerDn       string `yaml:"managerDn" xml:"managerDn"`
+	ManagerPassword string `yaml:"managerPassword" xml:"managerPassword"`
+}
+
+type LdapSettings struct {
+	LdapSettingArr []LdapSetting `yaml:"ldapSetting" xml:"ldapSetting"`
+}
+
+type xmlLdapConfig struct {
```

Comment:
Did you mean this to be lowercase? It will effect it's visibility

Reply:
Will correct it to upper case.